src/components/ViewOnly.js

```diff
@@ -32,115 +35,164 @@ class ViewOnly extends React.Component {
       notfound: false,
       originalCode: '',
     };
-    this.savePrevProgram = this.props.uid !== '';
-    this.props.setTheme(cookies.getThemeFromCookie());
+    this.savePrevProgram = uid !== '';
+    setTheme(cookies.getThemeFromCookie());
   }
 
-  componentDidMount = async () => {
+  async componentDidMount() {
+    const {
+      programid, mostRecentProgram, setProgramCode, setProgramLanguage, runCode,
+    } = this.props;
+
     if (this.savePrevProgram) {
       await this.codeSaverHelper();
     }
 
-    const { ok, sketch } = await fetch.getSketch(this.props.programid);
-
-    if (!ok) {
-      this.setState({ notfound: true });
-      return;
+    try {
```

Comment:
I'm not 100% sure, but I think all cases of bad things happening will be caught by the `.catch` function on the promise, so you don't need a separate try/catch system